packages/react-composites/src/composites/ChatComposite/file-sharing/FileCard.tsx

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import { Icon, mergeStyles, ProgressIndicator, Stack, Text, useTheme } from '@fluentui/react';
+import { getFileTypeIconProps } from '@fluentui/react-file-type-icons';
+import React from 'react';
+
+/**
+ * @beta
+ */
+export interface FileCardProps {
+  /**
+   * File name.
+   */
+  fileName: string;
+  /**
+   * Extension of the file used for rendering the file icon.
+   */
+  fileExtension: string;
+  /**
+   * File upload/download progress percentage between 0 and 1.
+   * Not shown if undefined or 1.
+   */
+  progress?: number;
+  /**
+   * Icon to display for actions like download, upload, etc. on the right of file name.
+   */
+  actionIcon?: JSX.Element;
+  /**
+   * Function that runs when actionIcon is clicked
+   */
+  actionHandler?: () => void;
+}
+
+/**
+ * @beta
+ */
+export const FileCard = (props: FileCardProps): JSX.Element => {
+  const { fileName, fileExtension, progress, actionIcon } = props;
+  const theme = useTheme();
+
+  return (
+    <Stack
+      className={mergeStyles({
+        width: '14rem',
+        background: theme.palette.neutralLighter,
+        borderRadius: theme.effects.roundedCorner4,
+        boxShadow: theme.effects.elevation8
+      })}
+    >
+      <Stack
+        horizontal
+        horizontalAlign="space-between"
+        verticalAlign="center"
+        className={mergeStyles({
+          padding: progress === undefined ? '0.75rem' : '0.75rem 0.75rem 0.5rem 0.75rem'
```

Comment:
Please define a couple style constants and pick between the two.

Also, please add a comment as to why extra padding is needed when progress bar is being shown. It's very hard to understand the reason for CSS properties after the fact.

Reply:
Optimized the code to make it less confusing instead :)